Add benchmarks for after-convert callbacks

rpuch opened the pull request:
- add the benchmark itself
- migrate to spring-boot dependencies of version 2.3.0
- add mvn wrapper for convenience as enforcer rules require a specific version of maven

This commit is still a WORK IN PROGRESS. A couple of problems for now:
- MongoResultsWriter is broken (code is just commented-out to make the project compile)
- spring-data-mongodb version is forced to be a snapshot one, not the one from spring-boot-dependencies BOM (so that it is possible to benchmark the needed version)

rpuch commented:
This pull request is related to https://github.com/spring-projects/spring-data-mongodb/pull/839

mp911de commented:
Thanks for your changes. Please remove the Maven Wrapper from the PR as it's not part of the actual change.

rpuch commented:
1. I've removed mvn wrapper
2. `MongoResultsWriter` has been ported to `net.minidev:json-smart` as `JSON` class has been removed from mongo driver
3. I've also removed spring-data-mongodb version override

You asked me to remove mvn wrapper. Should I add it in another PR, or it is not to be used here at all? The thing is that enforcer plugin requires a specific version of Maven (3.5.0) which is not terribly convenient without the usage of maven wrapper :)

rpuch commented:
I've added a PR with maven wrapper: https://github.com/spring-projects/spring-data-dev-tools/pull/141

mp911de commented:
Thanks a lot. That's merged and polished now.
